### PR TITLE
Add support for Server-Side RPC call.

### PR DESCRIPTION
**Pull request by lucky2you**
Add supoort for Server-Side RPC call.
Move MQTT client related code from MqttGatewayService.java to MqttGatewayClient.java

### CODE REVIEW

**Comment by ashvayka** [comment]
Hi @mp-loki! Please review this carefully. HI @lucky2you It would be great if you can create two PRs or at least two commits separately for your changes and for refactoring.

File: src/main/java/org/thingsboard/gateway/service/gateway/MqttGatewayClient.java [hunk 1]
```diff
@@ -0,0 +1,580 @@
...
+    private static final String GATEWAY_TELEMETRY_TOPIC = "v1/gateway/telemetry";
+    private static final String GATEWAY_REQUESTS_ATTRIBUTES_TOPIC = "v1/gateway/attributes/request";
+    private static final String GATEWAY_RESPONSES_ATTRIBUTES_TOPIC = "v1/gateway/attributes/response";
+    private static final String GATEWAY_CONNECT_TOPIC = "v1/gateway/connect";
+    private static final String GATEWAY_DISCONNECT_TOPIC = "v1/gateway/disconnect";
+    private static final String GATEWAY = "GATEWAY";
+
+    private static final String DEVICE_RPC_REQUEST_TOPIC = "v1/devices/me/rpc/request";
+    private static final String DEVICE_RPC_REQUEST_PLUS_TOPIC = "v1/devices/me/rpc/request/+";
+    private static final String DEVICE_RPC_RESPONSE_TOPIC = "v1/devices/me/rpc/response";
```
Comment (mp-loki) [inline src/main/java/org/thingsboard/gateway/service/gateway/MqttGatewayClient.java]:
```java
private static final String DEVICE_RPC_REQUEST_TOPIC = "v1/devices/me/rpc/request";
private static final String DEVICE_RPC_REQUEST_PLUS_TOPIC = "v1/devices/me/rpc/request/+";
private static final String DEVICE_RPC_RESPONSE_TOPIC = "v1/devices/me/rpc/response";
```
These are the topic subscriptions that are added. Are they intended to be used by Device-initiated RPC call? or Two-Way Server-Side RPC call? Also, there is no processing for these topics in onMessage() method

File: src/main/java/org/thingsboard/gateway/service/gateway/MqttGatewayClient.java [hunk 2]
```diff
@@ -0,0 +1,580 @@
...
+        callbackExecutor.submit(() -> {
+            try {
+                if (topic.equals(GATEWAY_ATTRIBUTES_TOPIC)) {
+                    onAttributesUpdate(message);
+                } else if (topic.equals(GATEWAY_RESPONSES_ATTRIBUTES_TOPIC)) {
+                    onDeviceAttributesResponse(message);
+                } else if (topic.equals(GATEWAY_RPC_TOPIC)) {
+                    onRpcCommand(message);
+                } else if (topic.startsWith(DEVICE_RPC_REQUEST_TOPIC)) {
+                    // TODO: add RPC command process
```
Comment (mp-loki) [inline src/main/java/org/thingsboard/gateway/service/gateway/MqttGatewayClient.java]: Please add processing for DEVICE_RPC_REQUEST_TOPIC, DEVICE_RPC_REQUEST_PLUS_TOPIC , DEVICE_RPC_RESPONSE_TOPIC

**Comment by zbeacon** [comment]
Hi, @lucky2you 

I am glad to inform you that we have released new version of the Gateway based on Python. This is a major rewrite and improvement from Java version. Main benefits:
Less footprint and memory impact
Easier to extend and customize
Improved functionality.

The new gateway supports RPC requests from ThingsBoard to external MQTT brokers with/without responses. You can try it instead of a Java gateway.

We look forward to your feedback.

**Pull request closed without merging**